test/e2e/framework/psp_util.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+	"sync"
+
+	rbacv1beta1 "k8s.io/api/rbac/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/onsi/ginkgo"
+)
+
+const (
+	PodSecurityPolicyPrivileged     = "privileged"
+	PodSecurityPolicyPrivilegedRole = "podsecuritypolicy:privileged"
+)
+
+var (
+	isPSPEnabledOnce sync.Once
+	isPSPEnabled     bool
+)
+
+func IsPodSecurityPolicyEnabled(f *Framework) bool {
+	isPSPEnabledOnce.Do(func() {
+		psp, err := f.ClientSet.ExtensionsV1beta1().PodSecurityPolicies().
```

Comment:
this makes the name of PSP act like an API, which isn't great. can the e2e test do `[Feature:PodSecurityPolicy]` instead?

Reply:
Then it won't run in the default test suite, which is what I was trying to avoid by doing it this way. The other alternative would be to pipe something through the `TestContext`. WDYT?

Reply:
hmm. If we want to detect enablement, presence of *any* PSP objects seems more reasonable to know if the feature is in use, then have the PSP-specific test create and clean up PSP objects specific to its test rather than depend on deployment-specific ones with well-known names.